### PR TITLE
Feature integrate pravega on vs7

### PR DESCRIPTION
**Pull request by AaronCOD**
## Purpose
[OBSDEF-4450](https://jira.cec.lab.emc.com:8443/browse/OBSDEF-4450)
_List the changes for this PR_
Add namespace for each cluster-scoped crds in pravega. 
Add validatingwebhookconfigurations CRD to the operaotor' role.
Add namespace-scoped watching for pravega and bk operators 
## PR checklist
- [x] make test passed
- [x] make build passed
- [x] helm install <chart> passed
- [x] vSphere7 make package deployments passed
- [x] helm upgrade <chart> passed
- [x] helm test <release_name> passed

## Testing
https://asd-ecs-jenkins.isus.emc.com/jenkins/job/ecs-flex-automation-custom-runner/2556/
https://asd-ecs-jenkins.isus.emc.com/jenkins/job/ecs-flex-automation-custom-runner/2555/

### CODE REVIEW

**Review by fedor-chemashkin — APPROVED** [review]

File: bookkeeper-operator/templates/webhook.yaml [hunk 1]
```diff
@@ -19,7 +19,7 @@ spec:
 apiVersion: admissionregistration.k8s.io/v1beta1
 kind: ValidatingWebhookConfiguration
 metadata:
-  name: bookkeeper-webhook-config
+  name: {{ .Release.Namespace }}-bookkeeper-webhook-config
```
Comment (eddiepavkovic) [inline bookkeeper-operator/templates/webhook.yaml:22]: missing namespace entry for this.

File: pravega-operator/templates/webhook.yaml [hunk 2]
```diff
@@ -19,7 +19,7 @@ spec:
 apiVersion: admissionregistration.k8s.io/v1beta1
 kind: ValidatingWebhookConfiguration
 metadata:
-  name: pravega-webhook-config
+  name: {{ .Release.Namespace }}-pravega-webhook-config
```
Comment (eddiepavkovic) [inline pravega-operator/templates/webhook.yaml:22]: missing `namespace:` entry for this resource.

File: bookkeeper-operator/templates/webhook.yaml (hunk 1 above)
Comment (AaronCOD) [reply bookkeeper-operator/templates/webhook.yaml:22]: Thanks for your reminding, the pravega original chart[https://github.com/pravega/bookkeeper-operator/blob/master/charts/bookkeeper-operator/templates/webhook.yaml#L22](url) also do not have namesapce entry. I think ValidatingWebhookConfiguration is a cluster-scoped resource so there is no need to have a namespace specified.

File: pravega-operator/templates/webhook.yaml (hunk 2 above)
Comment (AaronCOD) [reply pravega-operator/templates/webhook.yaml:22]: Same as the previous one.

File: bookkeeper-operator/templates/operator.yaml [hunk 3]
```diff
@@ -31,7 +31,7 @@ spec:
           {{- end }}
           env:
             - name: WATCH_NAMESPACE
-              value: {{ .Values.watchNamespace }}
+              value: "{{ .Release.Namespace }}"
```
Comment (eddiepavkovic) [inline bookkeeper-operator/templates/operator.yaml:34]: this isn't going to work right if we need to watch all namespaces.  or am i missing something here.

Comment (AaronCOD) [reply bookkeeper-operator/templates/operator.yaml:34]: The pravega and bk operator should only watch its own namespace becasue we will create operators in each namespace in VS7.

**Review by eddiepavkovic — APPROVED** [review]
conditionally approved,  please put OBSDEF-4450 in the summary above and in the PR so that we can track why these changes were made.

**Comment by AaronCOD** [comment]
This PR includes a workaround to unblock QE's testing with Pravega. Some changes in this pr will be removed when the final solution is delivered.

**Pull request merged**